File: diplomski-backend/src/main/java/com/example/diplomskibackend/model/Advertisement.java
package com.example.diplomskibackend.model;

import org.apache.el.lang.ELArithmetic;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
public class Advertisement {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column
    private String title;
    @Column
    private String text;
    @Column
    private Float price;
    @Column
    private String image;
    @Column
    private LocalDateTime startDate;
    @Column
    private LocalDateTime endDate;
    @Column
    private Boolean enable;
    @ManyToOne
    private Company company;

    public Advertisement() {
    }

    public Advertisement(String title, String text, Float price, String image, LocalDateTime startDate,  LocalDateTime endDate, Boolean enable,  Company company) {
        this.title = title;
        this.text = text;
        this.price = price;
        this.image = image;
        this.startDate = startDate;
        this.endDate = endDate;
        this.enable = enable;
        this.company = company;
    }



    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Float getPrice() {
        return price;
    }

    public void setPrice(Float price) {
        this.price = price;
    }

    public LocalDateTime getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDateTime startDate) {
        this.startDate = startDate;
    }

    public LocalDateTime getEndDate() {
        return endDate;
    }

    public void setEndDate(LocalDateTime endDate) {
        this.endDate = endDate;
    }

    public Company getCompany() {
        return company;
    }

    public void setCompany(Company company) {
        this.company = company;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public Boolean getEnable() {
        return enable;
    }

    public void setEnable(Boolean enable) {
        this.enable = enable;
    }
}

File: diplomski-frontend/src/app/model/advertisement.ts
interface AdvertisementInterface{
  id:number;
  title: string ;
  text: string ;
  price: number ;
  image: string;
  startDate: Date;
  endDate: Date;
  enable:boolean;
  companyId : number;

}
export class Advertisement implements AdvertisementInterface{
  id:number;
  title: string ;
  text: string ;
  price: number ;
  image: string;
  startDate: Date;
  endDate: Date;
  enable:boolean;
  companyId : number;

  constructor(obj:AdvertisementInterface){
      this.id=obj.id;
      this.title=obj.title;
      this.text=obj.text;
      this.price = obj.price;
      this.image=obj.image;
      this.startDate=obj.startDate;
      this.endDate=obj.endDate;
      this.enable=obj.enable;
      this.companyId=obj.companyId;
  }
}
